aten/src/ATen/cuda/CublasHandlePool.cpp

```diff
@@ -47,6 +87,13 @@ cublasHandle_t getCurrentCUDABlasHandle() {
   auto handle = myPoolWindow->reserve(device);
   auto stream = c10::cuda::getCurrentCUDAStream();
   TORCH_CUDABLAS_CHECK(cublasSetStream(handle, stream));
+#ifndef USE_ROCM
+  if (handle_to_workspace.find(handle) == handle_to_workspace.end()) {
+      auto workspace_ptr = getNewWorkspace();
+      handle_to_workspace[handle] = std::move(workspace_ptr);
+  }
+  TORCH_CUDABLAS_CHECK(cublasSetWorkspace(handle, handle_to_workspace[handle].get(), getChosenWorkspaceSize()));
```

Comment:
I don't think this would work correctly if the same handle is used on the different streams (which is legit use). Workspace will initially be allocated on the stream that is current, and we'd submit workload on it, then we can immediately change the stream, use the same handle (because that's what's supposed to happen, handles are per thread), it will find existing workspace and try to use it for an operation on the different stream at the same time.